### PR TITLE
use-value-compare errors when using radix number

### PR DESCRIPTION
**Pull request by StevenLooman**
use-value-compare errors when using radix number

### CODE REVIEW

**Comment by sebastiaanspeck** [comment]
Does it error when using 16e2/16e+2/16e-2/2&3/-10/-1.0/36rz as well?

**Comment by StevenLooman** [comment]
Actually, it appears that bignum starts at 1 << 31:

```
Magik> (1.shift(31) - 1).class_name
:integer 
Magik> 1.shift(31).class_name
:bignum 
```

Though I'm unsure about SW4. Do you have access to SW4, @sebastiaanspeck ?

**Comment by sebastiaanspeck** [comment]
> Actually, it appears that bignum starts at 1 << 31:
> 
> 
> 
> ```
> 
> Magik> (1.shift(31) - 1).class_name
> 
> :integer 
> 
> Magik> 1.shift(31).class_name
> 
> :bignum 
> 
> ```
> 
> 
> 
> Though I'm unsure about SW4. Do you have access to SW4, @sebastiaanspeck ?

I do,

``` 
MagikSF> (1.shift(31) - 1).class_name
$
:bignum 
MagikSF> 1.shift(31).class_name
$
:bignum 
```

**Comment by StevenLooman** [comment]
Thanks! The number that was present in the check (`1 <<29` or `536870912`) is the first value of a bignum.

File: magik-squid/src/main/java/nl/ramsolutions/sw/magik/api/MagikNumberParser.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+   * <p>Throws an exception when number could not be parsed.
+   *
+   * @param numberStr
+   * @return Parsed number.
+   */
+  public static Number parseMagikNumber(final String numberStr) {
+    final Number number;
+    if (numberStr.contains("r")) {
+      try (final Scanner scanner = new Scanner(numberStr)) {
+        scanner.useDelimiter("r");
```
Comment (sebastiaanspeck) [inline magik-squid/src/main/java/nl/ramsolutions/sw/magik/api/MagikNumberParser.java]:
```suggestion
    if (numberStr.contains("r") || numberStr.contains("R")) {
      try (final Scanner scanner = new Scanner(numberStr)) {
        scanner.useDelimiter("r|R");
```

**Pull request merged**